### PR TITLE
feat(oracle): Oracle that uses text values for feeds.

### PR DESCRIPTION
**Pull request by yordanmadzhunkov**
* Add an example with string.
* Update check script

Update config

Test string oracle

### CODE REVIEW

**Comment by yordanmadzhunkov** [comment]
```bash
hackaton$ ./check_btc_price.sh 
UpgradeableProxy contract address 0xc04b335A75C5Fa14246152178f6834E3eBc2DC7C
Anvil container id 128cc4d29a68
BTC Price from contarct
65994.8

Raw bytes from contarct
0x48656c6c6f20617765736f6d6520426c6f636b636861696e


Text from contarct
Hello awesome Blockchain
```

File: config/feeds_config.json [hunk 1]
```diff
@@ -66,7 +70,33 @@
...
+      "fullName": "Some text",
+      "description": "Hello World",
+      "decimals": 8,
+      "report_interval_ms": 30000,
+      "quorum_percentage": 1,
+      "type": "Forex",
+      "script": "YahooFinance",
+      "pair": {
+        "base": "EUR",
+        "quote": "USD"
```
Comment (EmilIvanichkovv) [inline config/feeds_config.json:89]: Can we remove this?

File: config/feeds_config.json [hunk 2]
```diff
@@ -66,7 +70,33 @@
...
+      },
+      "first_report_start_time": {
+        "secs_since_epoch": 0,
+        "nanos_since_epoch": 0
+      },
+      "resources": {
+        "yf_symbol": "EURUSD=X"
+      },
```
Comment (EmilIvanichkovv) [inline config/feeds_config.json:97]: ditto
Or maybe set `TEXTDATA` ?

File: examples/string_oracle/spin.toml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+id = "47"
+data = "ETHUSD"
+
+[[trigger.oracle.data_feeds]]
+id = "31"
+data = "BTCUSD"
+
+[[trigger.oracle.data_feeds]]
+id = "253"
+data = "EURUSD"
```
Comment (EmilIvanichkovv) [inline examples/string_oracle/spin.toml:27]: Remove?

File: examples/string_oracle/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+            let body = resp.into_body();
+            let string = String::from_utf8(body).expect("Our bytes should be valid utf8");
+            let value: Rate = serde_json::from_str(&string).unwrap();
+
+            println!("{:?}", value);
+
+            payload.values.push(DataFeedResult {
+                id: data_feed.id.clone(),
+                value: DataFeedResultValue::Numerical(value.rate),
+            });
```
Comment (EmilIvanichkovv) [inline examples/string_oracle/src/lib.rs]: Remove and just leave the `string` case?

**Review by EmilIvanichkovv — CHANGES_REQUESTED** [review]

File: examples/string_oracle/src/lib.rs (hunk 4 above)
Comment (yordanmadzhunkov) [reply examples/string_oracle/src/lib.rs]: I want to show that string and numerical feeds can be reported by the same oracle script

File: config/feeds_config.json (hunk 1 above)
Comment (yordanmadzhunkov) [reply config/feeds_config.json:89]: No, smart contract deployment expects some data here

**Pull request merged**